cms/templates/js/unit-access-editor.underscore

```diff
@@ -0,0 +1,133 @@
+<%
+var groupAccess = xblockInfo.get('group_access');
+var selectablePartitions = [];
+var groupIdToName = {};
+
+contentGroups.forEach(function (partition) {
+    if (partition['scheme'] != 'cohort' && partition['scheme'] != 'enrollment_track') {
+        return;
+    }
+    if (partition['scheme'] == 'enrollment_track') {
+        var tracks = [];
+        var trackSelected = false;
+        partition['groups'].forEach(function (group) {
+            if (group['selected']) {
+                trackSelected = true;
+            }
+        });
+        // Only add enrollment tracks if there are multiple tracks to restrict from
+        if (partition['groups'].length < 2 && !trackSelected) {
+            return;
+        }
+    }
+    selectablePartitions.push(partition);
+});
+
+selectablePartitions.forEach(function (partition) {
+    var schemeId = partition['id'];
+    var groups = partition['groups'];
+    groups.forEach(function (group) {
+        var groupId = group['id'];
+        var groupName = group['name'];
+        var fullId = schemeId.toString() + '-' + groupId.toString();
+        groupIdToName[fullId] = groupName;
+    })
+});
+
+var deletedGroups = [];
+
+for (var schemeId in groupAccess) {
+    groupAccess[schemeId].forEach(function (groupId) {
+        var fullId = schemeId.toString() + '-' + groupId.toString();
+        if (!groupIdToName[fullId]) {
+            deletedGroups.push(fullId);
+        }
+    })
+}
+
+// Need to include deleted groups logic to not change is_active boolean in contentstore/utils.py get_user_partition_info
```

Comment:
Let's not work around the implementation of get_user_partition_info. If a bug exists in it with regard to deleted groups not always showing, the fix should be in that method so that the component access editor on the unit page also picks up the fix.

If this is not trivial to fix, I'm fine with you creating a separate bug (assuming that the behavior you are describing shows up in both dialogs-- if it doesn't show up in both locations, I'd like to understand why).